File: ws/cañarte/u2/WS23UnitTesting/ComplexOperationsPython/ec_edu_espe_complexoperations_controller/operation.py
from decimal import Decimal, ROUND_HALF_UP


class operation:
    @staticmethod
    def add_two_numbers(addend1, addend2):
        bd1 = Decimal(str(addend1))
        bd2 = Decimal(str(addend2))
        sum_result = bd1 + bd2
        return float(sum_result.quantize(Decimal('0.00'), rounding=ROUND_HALF_UP))

File: ws/cañarte/u2/WS23UnitTesting/ComplexOperations/src/main/java/ec/edu/espe/complexoperation/controller/Operation.java
package ec.edu.espe.complexoperation.controller;

import java.math.BigDecimal;
import java.math.RoundingMode;

/**
 *
 * @author Saray Cañarte, codecreators, DCCO - ESPE
 */
public class Operation {

    public static float addTwoNumbers(float addend1, float addend2) {
        BigDecimal bd1 = BigDecimal.valueOf(addend1);
        BigDecimal bd2 = BigDecimal.valueOf(addend2);
        BigDecimal sum = bd1.add(bd2);
        return sum.setScale(2, RoundingMode.HALF_UP).floatValue(); 
    }
}

